src/include/type/value.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "common/typedefs.h"
+#include "type/type_id.h"
+
+namespace terrier::type {
+/**
+ * A Value is a tagged union of a SQL type ID and the contents.
+ */
+struct Value {
```

Comment:
This Value placeholder will likely change. I have a simple implementation of Value that I will submit into this PR. It can better evaluated in the context of the expression code.